src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static final double CAPACITY_EXPAND_FACTOR = 1.5;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[CAPACITY];
+    }
+
+    public ArrayList(List<T> list) {
+        objects = new Object[CAPACITY];
+        if (list.size() <= CAPACITY) {
+            copyListToArrayListAndResize(list);
+            return;
+        }
+        do {
+            expandObjectsArray(objects);
+        } while (list.size() <= CAPACITY);
+        copyListToArrayListAndResize(list);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < CAPACITY) {
+            objects[size] = value;
+            size++;
```

Comment:
you have the same lines below, rewrite your logic, so you could remove duplicates

Reply:
Thanks, done.